eventlet/support/greendns.py

```diff
@@ -513,6 +513,8 @@ def getaddrinfo(host, port, family=0, socktype=0, proto=0, flags=0):
     """
     if isinstance(host, six.string_types):
         host = host.encode('idna').decode('ascii')
+    elif isinstance(host, bytes):
+        host = host.decode('ascii')
```

Comment:
The fix makes sense -- if the host is bytes, assume it's already IDNA-encoded -- but I wonder if this is the right place for it. Fixing `getaddrinfo` but not `resolve` or `ResolverProxy.query` or `HostsResolver.query` seems strange

Reply:
I agree with @tipabu resolver should be fixed too